Sources/MapboxDirections/DirectionsResult.swift

```diff
@@ -62,6 +63,8 @@ open class DirectionsResult: Codable {
         } else {
             speechLocale = nil
         }
+
+        speechLocaleIsSupported = container.contains(.speechLocale)
```

Comment:
Based on this logic, I think `speechLocaleIsSupported` should be renamed. It’s true if and only if `voiceLocale` is present in the JSON. In other words, it’s true as long as voice instructions were requested, whether or not the language is supported. If the language is unsupported, this property would be `false` but `speechLocale` would be `nil`.

Basically, we need to distinguish between “no speech locale” and “speech locale not applicable”, at least when it comes to the Mapbox Directions API. I can’t think of a great name for the property at the moment. One option would be to remove this property in favor of looking at whether the route options has `includesSpokenInstructions` set to `true`, but we should make sure there aren’t other cases where `voiceInstructions=true` in the request but `voiceLocale` is undefined in the response.

/cc @mandeepsandhu @danpaz